### DISCUSSION TITLE
Reconsider material development workflow

### DISCUSSION THREAD

**Discussion by QbieShay** [godotengine/godot-proposals#5235] [Editor]
### Describe the project you are working on

Various VFX

### Describe the problem or limitation you are having in your project

Since Godot 4 has per-instance uniforms, tweaking a material that you're working on became a bit clunky: properties are split between the material resource and the instance. Furthermore, changes in the way resources are focused makes it challenging to have the shader open at the same time as the material. There's a lot of clicking and objects losing focus.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I don't have a solution to this. I'm open to any solution that would make me lose less track of what I'm doing.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

See above

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Its about the editor

### Is there a reason why this should be core and not an add-on in the asset library?

See above

**Comment by Calinou** [comment]
Moved to discussion, as there is no concrete implementation proposal.

**Reply by QbieShay** [reply]
Oops, thank you!

**Comment by Zireael07** [comment]
Hmm, I think it could be solved by having a separate window (akin to advanced import/fix dependencies) that lets you see both at once. Being a window/popup it wouldn't have focus problems.

WDYT?
